### PR TITLE
Fix for trivia not preserved after string interpolation conversion

### PR DESCRIPTION
**Pull request by AppAppWorks**
fix an issue in #1551

My bad, I should've tested the code action more thoroughly. The leading trivia of the source `SequenceExprSyntax` is not carried over to the new string literal, resulting in this:

```swift
<code block A>
"" + value + ""
<code block B>
```
->
```swift
<code block A>"\(value)"
<code block B>
```

This fix will preserve the leading trivia, i.e.
```swift
<code block A>
"\(value)"
<code block B>
```

### CODE REVIEW

**Comment by ahoppen** [comment]
How was `// code block A` preserved before this change if the leading trivia of the code block weren’t preserved? Or did you mean for `// code block A` to be a statement?

Should we also retain the trailing trivia from `syntax` in the same way?

**Comment by AppAppWorks** [comment]
Yeah,  `// code block A` was meant for a statement instead of a comment, I've changed that in the description above.

AFAIK, string literals don't have trailing trivia, and trailing trivia of any non-string-literal component will be converted and moved into the string interpolation, so the trailing trivia of `syntax` has been preserved in this way.

**Comment by ahoppen** [comment]
What about `"a" + b + "c" / * some comment */`? In that case `/* some comment */` should be trailing trivia for the `"`.

**Review by ahoppen — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: Tests/SourceKitLSPTests/CodeActionTests.swift [hunk 1]
```diff
@@ -1022,9 +1024,10 @@ final class CodeActionTests: XCTestCase {
             changes: [
               uri: [
                 TextEdit(
-                  range: positions["1️⃣"]..<positions["5️⃣"],
+                  range: positions["0️⃣"]..<positions["5️⃣"],
                   newText: ###"""
-                    ##"[\##(key): \##(d) \##(value)]"##
+
```
Comment (ahoppen) [inline Tests/SourceKitLSPTests/CodeActionTests.swift:1029]: Looks like this line has trailing whitespace. Could you run swift-format on your changes? https://github.com/swiftlang/sourcekit-lsp/blob/main/CONTRIBUTING.md#formatting

Comment (AppAppWorks) [reply Tests/SourceKitLSPTests/CodeActionTests.swift]: Done, please test again.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**